src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,204 @@
+            this.value = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (first == null) {
+            linkFirst(value);
+            return;
+        }
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndex(index);
+        if (index == 0) {
+            linkFirst(value);
+            return;
+        }
+        if (index == size) {
+            linkLast(value);
+            return;
+        } else {
+            linkBefore(value, node(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        Object[] listToAdd = list.toArray();
+        int countToAdd = listToAdd.length;
+
+        for (int i = 0; i < countToAdd; i++) {
+            if (first == null) {
+                linkFirst((T) listToAdd[i]);
+            } else {
+                linkLast((T) listToAdd[i]);
+            }
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkElementIndex(index);
+        return node(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> nodeToSet = node(index);
+        T oldVal = nodeToSet.value;
+        nodeToSet.value = value;
+        return oldVal;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkElementIndex(index);
+        return unlink(node(index));
     }
 
     @Override
     public boolean remove(T object) {
+        if (object == null) {
+            for (Node<T> x = first; x != null; x = x.next) {
+                if (x.value == null) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> x = first; x != null; x = x.next) {
+                if (object.equals(x.value)) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
+    }
+
+    private T unlink(Node<T> nodeToErase) {
+        final T removed = nodeToErase.value;
+        final Node<T> next = nodeToErase.next;
+        final Node<T> prev = nodeToErase.prev;
+        if (prev == null) {
+            first = next;
+        } else {
+            prev.next = next;
+            nodeToErase.prev = null;
+        }
+        if (next == null) {
+            last = prev;
+        } else {
+            next.prev = prev;
+            nodeToErase.next = null;
+        }
+        nodeToErase.value = null;
+        size--;
+        return removed;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return (first == null);
+    }
+
+    private Node<T> node(int index) {
```

Comment:
method names should start from verbs https://mate-academy.github.io/style-guides/java/java.html#s5.2.3-method-names